code/components/citizen-devtools/src/ResourceTimeWarnings.cpp

```diff
@@ -67,7 +69,67 @@ constexpr auto tuple_slice(Cont&& t)
 	return detail::slice_impl<I1>(std::forward<Cont>(t),
 	std::make_index_sequence<I2 - I1>{});
 }
+
+static std::chrono::microseconds lastHitch;
+
+#ifdef GTA_FIVE
+#include <Hooking.h>
+#include <InputHook.h>
+
+static decltype(&SetThreadExecutionState) origSetThreadExecutionState;
+static decltype(&PeekMessageW) origPeekMessageW;
+
+static std::chrono::microseconds trackTotal;
+static bool trackIsInvalid = false;
+
+static EXECUTION_STATE WINAPI SetThreadExecutionState_Track(EXECUTION_STATE esFlags)
+{
+	if (esFlags == 3)
+	{
+		trackTotal = std::chrono::microseconds{ 0 };
+		trackIsInvalid = false;
+	}
+
+	return origSetThreadExecutionState(esFlags);
+}
+
+static BOOL WINAPI PeekMessageW_Track(LPMSG lpMsg, HWND hWnd, UINT wMsgFilterMin, UINT wMsgFilterMax, UINT wRemoveMsg)
+{
+	if (trackIsInvalid)
+	{
+		return origPeekMessageW(lpMsg, hWnd, wMsgFilterMin, wMsgFilterMax, wRemoveMsg);
+	}
+
+	auto trackStart = usec();
+	auto rv = origPeekMessageW(lpMsg, hWnd, wMsgFilterMin, wMsgFilterMax, wRemoveMsg);
+	auto trackEnd = usec();
+	trackTotal += (trackEnd - trackStart);
+
+	if (!rv && !trackIsInvalid)
+	{
+		if (trackTotal > 30ms)
+		{
+			lastHitch = trackEnd;
+		}
+	}
+
+	return rv;
+}
+
+static HookFunction hookFunctionGameTime([]()
+{
+	InputHook::DeprecatedOnWndProc.Connect([](HWND, UINT uMsg, WPARAM, LPARAM, bool&, LRESULT&)
+	{
+		if (uMsg == WM_ACTIVATEAPP)
+		{
+			trackIsInvalid = true;

```

Comment:
An attempt to reduce any time taken when alt-tabbing in. Maybe dead code, should be checked.